cmd/editorconfig-checker/main_test.go

```diff
@@ -133,6 +135,68 @@ func TestMainDryRun(t *testing.T) {
 	}
 }
 
+// helper to set a map of environment variables and return a function to use in t.Cleanup to reset once the test completed
+// thanks to https://dev.to/arxeiss/auto-reset-environment-variables-when-testing-in-go-5ec
+func envSetter(envs map[string]string) (closer func()) {
+	originalEnvs := map[string]string{}
+
+	for name, value := range envs {
+		if originalValue, ok := os.LookupEnv(name); ok {
+			originalEnvs[name] = originalValue
+		}
+		_ = os.Setenv(name, value)
+	}
+
+	return func() {
+		for name := range envs {
+			origValue, has := originalEnvs[name]
+			if has {
+				_ = os.Setenv(name, origValue)
+			} else {
+				_ = os.Unsetenv(name)
+			}
+		}
+	}
+}
```

Comment:
This should be preferred, the error you ignore will get caught, and it would avoid issue on testes ran in parallel 

```suggestion
func envSetter(t *testing.T, envs map[string]string) (closer func()) {
	originalEnvs := map[string]string{}

	for name, value := range envs {
		if originalValue, ok := os.LookupEnv(name); ok {
			originalEnvs[name] = originalValue
		}
		t.Setenv(name, value)
	}

	return func() {
		for name := range envs {
			origValue, has := originalEnvs[name]
			if has {
				t.Setenv(name, origValue)
			} else {
				t.Unsetenv(name)
			}
		}
	}
}
```

Reply:
hm.. looking at `t.Setenv` - this means we can thin the helper much more, since it automatically does the cleanup.

Reply:
That's its purpose yes